### ISSUE TITLE
[Bug]: [bulk insert v2]standalone oom during bulk insert

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#31545]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.4-20240322-91749583-amd64
- Deployment mode(standalone or cluster):standalone
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

```
Containers:
  standalone:
    Container ID:  containerd://89d67e73b31f01184e9d22b098d2fa88c50b592af4fbaff44411ede94b03b7e1
    Image:         harbor.milvus.io/milvus/milvus:2.4-20240322-91749583-amd64
    Image ID:      harbor.milvus.io/milvus/milvus@sha256:e8d506782773c1c2eb2f20042a0f0b481f2f29745380f203c4ba420b4d281a46
    Ports:         19530/TCP, 9091/TCP
    Host Ports:    0/TCP, 0/TCP
    Args:
      /milvus/tools/run-helm.sh
      milvus
      run
      standalone
    State:          Running
      Started:      Fri, 22 Mar 2024 09:47:32 +0000
    Last State:     Terminated
      Reason:       OOMKilled
      Exit Code:    137
      Started:      Fri, 22 Mar 2024 09:20:58 +0000
      Finished:     Fri, 22 Mar 2024 09:47:31 +0000
    Ready:          True
    Restart Count:  1
    Limits:
      cpu:     2
      memory:  9Gi
    Requests:
      cpu:        1800m
      memory:     8294Mi
    Liveness:     http-get http://:metrics/healthz delay=90s timeout=5s period=30s #success=1 #failure=5
    Readiness:    http-get http://:metrics/healthz delay=90s timeout=5s period=10s #success=1 #failure=5
    Environment:  <none>
```

```
[2024-03-22T09:50:01.310Z] + grep bulk-insert-bench-test-21

[2024-03-22T09:50:01.311Z] + kubectl get pods -o wide

[2024-03-22T09:50:01.566Z] bulk-insert-bench-test-21-etcd-0                                  1/1     Running             0               29m     10.104.25.182   4am-node30   <none>           <none>

[2024-03-22T09:50:01.566Z] bulk-insert-bench-test-21-etcd-1                                  1/1     Running             0               29m     10.104.28.34    4am-node33   <none>           <none>

[2024-03-22T09:50:01.566Z] bulk-insert-bench-test-21-etcd-2                                  1/1     Running             0               29m     10.104.34.90    4am-node37   <none>           <none>

[2024-03-22T09:50:01.566Z] bulk-insert-bench-test-21-milvus-standalone-f8fcfc4f7-8nk5t       1/1     Running             1 (2m30s ago)   29m     10.104.18.72    4am-node25   <none>           <none>

[2024-03-22T09:50:01.566Z] bulk-insert-bench-test-21-minio-848459488d-mw97z                  1/1     Running             0               29m     10.104.31.138   4am-node34   <none>           <none>

[2024-03-22T09:50:01.566Z] bulk-insert-bench-test-21-t480r-9hpcr-2g4z2                       2/2     Running             0               32m     10.104.16.214   4am-node21   <none>           <none>
```

### Milvus Log

failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/bulk_insert_bench_test/detail/bulk_insert_bench_test/21/pipeline
log:

[artifacts-bulk-insert-bench-test-21-server-logs.tar.gz](https://github.com/milvus-io/milvus/files/14721669/artifacts-bulk-insert-bench-test-21-server-logs.tar.gz)

bulk insert info
```
[2024-03-22T09:49:57.688Z] [2024-03-22 09:47:25 - DEBUG - ci_test]: 

[2024-03-22T09:49:57.688Z] method: post, 

[2024-03-22T09:49:57.688Z] url: http://10.255.122.89:19530/v2/vectordb/jobs/import/get_progress, 

[2024-03-22T09:49:57.688Z] cost time: 0.02533888816833496, 

[2024-03-22T09:49:57.688Z] header: {'Content-Type': 'application/json', 'Authorization': 'Bearer :', 'RequestId': '3067230e-e831-11ee-b07e-4aa20de0a95c'}, 

[2024-03-22T09:49:57.688Z] payload: {"jobId": "448554419328370701"}, 

[2024-03-22T09:49:57.688Z] response: {"code":200,"data":{"collectionName":"bulk_insert_soChl5fm","details":[{"completeTime":"","fileName":"id:448554419328370702 paths:\"data-fields-9-rows-4710000-dim-128-file-num-1-error-none-1711100175.parquet\" ","fileSize":14645841801,"progress":0}],"jobId":"448554419328370701","progress":50,"state":"Importing"}} (test_bulk_insert_bench.py:37)
```

**Comment by zhuwenxing** [comment]
it works well for 2.3.
I did not change the resource limit and data size

**Comment by xiaofan-luan** [comment]
this might be due the cconcurrency control of bulkinsert V2.
@zhuwenxing do we test on the performance of bulkinsertV2 and V1 as well

**Comment by zhuwenxing** [comment]
/assign @bigsheeper 
PTAL

**Comment by zhuwenxing** [comment]
@xiaofan-luan 
> this might be due the cconcurrency control of bulkinsert V2.

This test has only one task, and this task has only one file. Based on my understanding, there should be no concurrency

> do we test on the performance of bulkinsertV2 and V1 as well

working on it

**Comment by bigsheeper** [comment]
related to: https://github.com/milvus-io/milvus/issues/31640

**Comment by bigsheeper** [comment]
/assign @zhuwenxing 
please help to verify

**Comment by zhuwenxing** [comment]
verified and fixed with image tag `2.4-20240401-4f3bfe24a-amd64`

![image](https://github.com/milvus-io/milvus/assets/12268675/da28bae3-5e45-4fbd-8726-b58bb2a61fea)
